public/js/maho-captcha.js

```diff
@@ -0,0 +1,178 @@
+/**
+ * Maho
+ *
+ * @package     Maho_Captcha
+ * @copyright   Copyright (c) 2025 Maho (https://mahocommerce.com)
+ * @license     https://opensource.org/licenses/afl-3.0.php  Academic Free License (AFL 3.0)
+ */
+
+const MahoCaptcha = {
+    altchaWidget: null,
+    altchaState: null,
+    frontendSelectors: '',
+    onVerifiedCallback: null,
+
+    async setup(config) {
+        this.altchaWidget = document.querySelector('altcha-widget');
+        this.frontendSelectors = config.frontendSelectors ?? '';
+
+        if (document.readyState === 'loading') {
+            document.addEventListener('DOMContentLoaded', this.initForms.bind(this));
+        } else {
+            this.initForms();
+        }
+    },
+
+    initForms() {
+        for (const formEl of document.querySelectorAll(this.frontendSelectors)) {
+            formEl.addEventListener('focusin', this.loadAltchaScripts.bind(this), true);
+            formEl.addEventListener('submit', this.onFormSubmit.bind(this), true);
+        }
+    },
+
+    async loadAltchaScripts() {
+        if (typeof customElements.get('altcha-widget') === 'function') {
+            return;
+        }
+
+        // Load Altcha JS
+        await Promise.all([
+            new Promise((resolve, reject) => {
+                const script = document.createElement('script');
+                script.src = '/js/altcha.min.js';
+                script.type = 'module';
+                script.onload = resolve;
+                script.onerror = () => reject(`${script.src} Not Found`);
+                document.head.appendChild(script);
+            }),
+        ]);
+
+        // Inject Altcha Widget
+        if (!this.altchaWidget) {
+            if (typeof config.widgetAttributes !== 'object' || config.widgetAttributes === null) {
+                throw new Error('widgetAttributes must be specified');
+            }
+            this.altchaWidget = document.createElement('altcha-widget');
+            for (const [ attr, val ] of Object.entries(config.widgetAttributes)) {
+                this.altchaWidget.setAttribute(attr, val);
+            }
+            document.body.appendChild(this.altchaWidget);
+        }
```

Comment:
if I'm not missing something, this should not be necessary since the `<altcha-widget` is in the phtml itself

Reply:
That's correct, it was just to support how you were creating the element before. However if in the other comment thread you were talking about putting the template output in the `<head>` element then we'd want to not output `<altcha-widget>` in the phtml and instead use JS.